modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/ReadOnlyUnbackedObservableList.java

```diff
@@ -185,8 +185,9 @@ public void callObservers(Change<E> c) {
 
     @Override
     public Object[] toArray() {
-        Object[] arr = new Object[size()];
-        for (int i = 0; i < size(); i++) {
+        int max = size();
+        Object[] arr = new Object[max];
+        for (int i = 0; i < max; i++) {
             arr[i] = get(i);
         }
         return arr;
```

Comment:
Have you tried `return stream().toArray();`?

Reply:
I haven't measured how `stream().toArray()` compare to a for..loop, but I inherently try to avoid streams in such scenarios.